pkg/helm/charts.go

```diff
@@ -0,0 +1,283 @@
+package helm
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path"
+	"sort"
+
+	helmv2beta1 "github.com/fluxcd/helm-controller/api/v2beta1"
+	sourcev1beta1 "github.com/fluxcd/source-controller/api/v1beta1"
+	pb "github.com/weaveworks/weave-gitops/pkg/api/profiles"
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chart"
+	"helm.sh/helm/v3/pkg/chart/loader"
+	"helm.sh/helm/v3/pkg/cli"
+	"helm.sh/helm/v3/pkg/getter"
+	"helm.sh/helm/v3/pkg/repo"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+)
+
+// ProfileAnnotation is the annotation that Helm charts must have to indicate
+// that they provide a Profile.
+const ProfileAnnotation = "weave.works/profile"
+
+// NewRepoManager creates and returns a new RepoManager.
+func NewRepoManager(kc client.Client, cacheDir string) *RepoManager {
+	return &RepoManager{
+		Client:   kc,
+		CacheDir: cacheDir,
+		envSettings: &cli.EnvSettings{
+			Debug:            true,
+			RepositoryCache:  cacheDir,
+			RepositoryConfig: path.Join(cacheDir, "/repository.yaml"),
+		},
+	}
+}
+
+// RepoManager implements HelmRepoManager interface using the Helm library packages.
+type RepoManager struct {
+	client.Client
+	CacheDir    string
+	envSettings *cli.EnvSettings
+}
+
+// ChartReference is a Helm chart reference, the SourceRef is a Flux
+// SourceReference for the Helm chart.
+type ChartReference struct {
+	Chart     string
+	Version   string
+	SourceRef helmv2beta1.CrossNamespaceObjectReference
+}
+
+// DefaultChartGetter provides default ways to get a chart index.yaml based on
```

Comment:
```suggestion
// DefaultChartGetters provides default ways to get a chart index.yaml based on
```

Reply:
On second thought. Why is this exported?

Reply:
indeed, fixing